lib/teleterm/vnet/service.go

```diff
@@ -245,6 +282,25 @@ func (p *appProvider) GetVnetConfig(ctx context.Context, profileName, leafCluste
 	return vnetConfig, trace.Wrap(err)
 }
 
+// OnNewConnection submits a usage event once per appProvider lifetime.
+// That is, if a user makes multiple connections to a single app, OnNewConnection submits a single
+// event. This is to mimic how Connect submits events for its app gateways. This lets us compare
+// popularity of VNet and app gateways.
+func (p *appProvider) OnNewConnection(ctx context.Context, profileName, leafClusterName string, app types.Application) error {
+	// Enqueue the event from a separate goroutine since we don't care about errors anyway and we also
+	// don't want to slow down VNet connections.
+	go func() {
+		uri := uri.NewClusterURI(profileName).AppendLeafCluster(leafClusterName).AppendApp(app.GetName())
+
+		err := p.usageReporter.ReportApp(ctx, uri)
```

Comment:
you might want to use a different context in case this is a very short-lived tcp connection and the context expires very quickly

Reply:
Hmm yeah, I just realized that in a regular local proxy, the context that is passed is the context behind the local proxy, while here it's the context behind the connection.

What's the best practice here? My first thought was a background context with an arbitrary timeout, but it feels like it'd need more coordination, e.g., the usage reporter should cancel all such contexts when VNet is getting shut down.

Reply:
I'm thinking of adding a WaitGroup and a `close` channel. VNet service would call `usageReported.Close` before exiting, this in turn would close the `close` channel and wait for the WaitGroup. Closing `close` would cancel all contexts.

Reply:
I didn't actually need a WaitGroup. I forgot that only one `ReportApp` can be active at a time, so there's ever only one context to cancel.